File: src/main/java/com/sdet/pages/BasePage.java
package com.sdet.pages;

import com.microsoft.playwright.Page;

public class BasePage {

    protected Page page;

    public BasePage(Page page) {
        this.page = page;
    }

    // ✅ Navigate to URL
    public void goToUrl(String url) {
        page.navigate(url);
    }

    // ✅ Click element
    public void click(String selector) {
        page.click(selector);
    }

    // ✅ Type text
    public void fill(String selector, String text) {
        page.fill(selector, text);
    }

    // ✅ Get text
    public String getText(String selector) {
        return page.textContent(selector);
    }

    // ✅ Check if visible
    public boolean isVisible(String selector) {
        return page.isVisible(selector);
    }

    // ✅ Get page title
    public String getTitle() {
        return page.title();
    }

    // ✅ Take screenshot
    public void takeScreenshot(String path) {
        page.screenshot(new Page.ScreenshotOptions()
            .setPath(java.nio.file.Paths.get(path)));
    }
}

File: pages/BasePage.ts
import { Page } from "@playwright/test";

export class BasePage {
    protected page: Page;
    constructor(page: Page) {
        this.page = page;
    }
    async goToUrl(url: string): Promise<void> {
        await this.page.goto(url);
    }
    async click(selector: string): Promise<void> {
        await this.page.locator(selector).click();
    }
    async fill(selector: string, value: string): Promise<void> {
        await this.page.locator(selector).fill(value);
    }
    async getText(selector: string): Promise<string> {
        return await this.page.locator(selector).innerText();
    }

    async isVisible(selector: string): Promise<boolean> {
        return await this.page.locator(selector).isVisible();
    }

    async getTitle(): Promise<string> {
        return await this.page.title();
    }

    async takeScreenshot(name: string): Promise<void> {
        await this.page.screenshot({ path: `screenshots/${name}.png` });
    }
}